### DISCUSSION TITLE
check if OpticStudio failed silently

### DISCUSSION THREAD

**Discussion by jwmbeenakker** [MREYE-LUMC/ZOSPy#47] [Ideas]
At the [OpticStudio forum](https://community.zemax.com/zos-api-12/please-throw-errors-instead-of-failing-silently-707?postid=14467#post14467) there is a discussion on the fact that the API doesn't trow exceptions if it fails.
The `TheSystem.GetCurrentStatus()` and associated logging functions might be something the add, probably in a more general way so it can be easily added to custom analyses (and maybe combine it with #20 and #34 )

**Comment by crnh** [comment]
We are currently researching this feature, but many errors are not reported by OpticStudio. [This has also been reported by another user at the OpticStudio forum.](https://community.zemax.com/zos-api-12/please-throw-errors-instead-of-failing-silently-707?postid=14473#post14473) Depending on our findings, we may choose not to implement features related to OpticStudio's status checking, as its added value seems to be rather limited.
